### PR TITLE
feat: favorite feature table icons

### PR DESCRIPTION
**Pull request by Tymek**
## About the changes
image

Closes [linear 1-403](https://linear.app/unleash/issue/1-403/frontend-add-ui-for-favoriting-features-and-projects)

### CODE REVIEW

File: frontend/src/component/feature/FeatureToggleList/FeatureToggleListTable.tsx [hunk 1]
```diff
@@ -126,7 +63,118 @@ export const FeatureToggleListTable: VFC = () => {
...
+                                  onClick={onTogglePinFavorites}
+                              />
+                          ),
+                          accessor: 'favorite',
+                          Cell: ({ row: { original: feature } }: any) => (
+                              <FavoriteIconCell
+                                  value={feature?.favorite}
+                                  onClick={() =>
+                                      feature?.favorite
+                                          ? removeFavoriteFeature(
```
Comment (kwasniew) [inline frontend/src/component/feature/FeatureToggleList/FeatureToggleListTable.tsx:94]: unfavorite

File: frontend/src/component/feature/FeatureToggleList/FeatureToggleListTable.tsx [hunk 2]
```diff
@@ -126,7 +63,118 @@ export const FeatureToggleListTable: VFC = () => {
...
+                                                feature.project,
+                                                feature.name
+                                            )
+                                          : addFavoriteFeature(
```
Comment (kwasniew) [inline frontend/src/component/feature/FeatureToggleList/FeatureToggleListTable.tsx:98]: favorite

Comment (kwasniew) [reply frontend/src/component/feature/FeatureToggleList/FeatureToggleListTable.tsx:98]: favorite(project, featureName)

File: frontend/src/hooks/usePinnedFavorites.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
+import { useMemo, useState } from 'react';
+import { sortTypes } from 'utils/sortTypes';
+
+const sortTypesWithFavorites = Object.fromEntries(
```
Comment (kwasniew) [inline frontend/src/hooks/usePinnedFavorites.ts:4]: unit test to describe how to use it

File: frontend/src/hooks/usePinnedFavorites.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    Object.entries(sortTypes).map(([key, value]) => [
+        key,
+        (v1: any, v2: any, id: string, desc?: boolean) => {
```
Comment (kwasniew) [inline frontend/src/hooks/usePinnedFavorites.ts:7]: I'd add minimum type expectations here

**Review by kwasniew — APPROVED** [review]
Non blocking approve but please check comments

File: frontend/src/component/feature/FeatureToggleList/FavoriteIconCell/FavoriteIconCell.tsx [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+            elseShow={
+                <IconButton
+                    onClick={onClick}
+                    size="small"
+                    sx={{
+                        color: 'transparent',
+                        '&:hover, &:focus': {
+                            color: theme => theme.palette.primary.main,
+                        },
+                    }}
```
Comment (nunogois) [inline frontend/src/component/feature/FeatureToggleList/FavoriteIconCell/FavoriteIconCell.tsx:35]: Maybe we should make this a `StyledIconButton`? A cell component with `sx` inside seems like something we shouldn't do in terms of performance.

Comment (Tymek) [reply frontend/src/component/feature/FeatureToggleList/FavoriteIconCell/FavoriteIconCell.tsx:35]: Good catch! Thanks

**Review by sjaanus — APPROVED** [review]
LGTM!

**Pull request merged**